src/Dynatrace.OpenTelemetry.Exporter.Metrics.Tests/Utils/NormalizeTests.cs

```diff
@@ -0,0 +1,222 @@
+        public void DimensionKeyNormalizedCorrectly(string name, string input, string expected)
+        {
+            Assert.Equal(expected, Normalize.DimensionKey(input));
+        }
+
+        [Fact]
+        public void DimensionKeyTruncatedCorrectly()
+        {
+            Assert.Equal(new string('a', 100), Normalize.DimensionKey(new string('a', 110)));
+        }
+
+        [Theory]
+        [InlineData("valid value", "value", "value")]
+        [InlineData("valid empty", "", "")]
+        [InlineData("pass null", null, "")]
+        [InlineData("valid uppercase", "VALUE", "VALUE")]
+        [InlineData("valid colon", "a:3", "a:3")]
+        [InlineData("valid value 2", "~@#ä", "~@#ä")]
+        [InlineData("valid spaces", "a b", "a b")]
+        [InlineData("valid comma", "a,b", "a,b")]
+        [InlineData("valid equals", "a=b", "a=b")]
+        [InlineData("valid backslash", "a\\b", "a\\b")]
+        [InlineData("valid multiple special chars", " ,=\\", " ,=\\")]
+        [InlineData("valid key-value pair", "key=\"value\"", "key=\"value\"")]
+        //     \u0000 NUL character, \u0007 bell character
+        [InlineData("invalid unicode", "\u0000a\u0007", "_a_")]
+        [InlineData("invalid unicode space", "a\u0001b", "a_b")]
+        //     'Ab' in unicode:
+        [InlineData("valid unicode", "\u0034\u0066", "\u0034\u0066")]
+        //     A umlaut, a with ring, O umlaut, U umlaut, all valid.
+        [InlineData("valid unicode", "\u0132_\u0133_\u0150_\u0156", "\u0132_\u0133_\u0150_\u0156")]
+        [InlineData("invalid leading unicode NUL", "\u0000a", "_a")]
+        [InlineData("invalid only unicode", "\u0000\u0000", "_")]
+        [InlineData("invalid consecutive leading unicode", "\u0000\u0000\u0000a", "_a")]
+        [InlineData("invalid consecutive trailing unicode", "a\u0000\u0000\u0000", "a_")]
+        [InlineData("invalid trailing unicode NUL", "a\u0000", "a_")]
+        [InlineData("invalid enclosed unicode NUL", "a\u0000b", "a_b")]
+        [InlineData("invalid consecutive enclosed unicode NUL", "a\u0000\u0007\u0000b", "a_b")]
+        public void DimensionValueNormalizedCorrectly(string name, string input, string expected)
+        {
+            Assert.Equal(expected, Normalize.DimensionValue(input));
+        }
+
+        [Fact]
+        public void DimensionValueTruncatedCorrectly()
+        {
+            Assert.Equal(new string('a', 250), Normalize.DimensionValue(new string('a', 270)));
+        }
+
+        [Theory]
+        [InlineData("escape spaces", "a b", "a\\ b")]
+        [InlineData("escape comma", "a,b", "a\\,b")]
+        [InlineData("escape equals", "a=b", "a\\=b")]
+        [InlineData("escape backslash", "a\\b", "a\\\\b")]
+        [InlineData("escape multiple special chars", " ,=\\", "\\ \\,\\=\\\\")]
+        [InlineData("escape consecutive special chars", "  ,,==\\\\", "\\ \\ \\,\\,\\=\\=\\\\\\\\")]
+        [InlineData("escape key-value pair", "key=\"value\"", "key\\=\"value\"")]
```

Comment:
`"` should be escaped I think

```suggestion
        [InlineData("escape key-value pair", "key=\"value\"", "key\\=\\"value\\"")]
```